components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -3369,7 +3369,7 @@ private boolean isPrimaryDomain(String newGroupDomainName) {
 
     private boolean isGroupDisplayNameChanged(String oldGroupDisplayName, String newGroupDisplayName) {
 
-        return !oldGroupDisplayName.equalsIgnoreCase(newGroupDisplayName);
+        return !oldGroupDisplayName.equals(newGroupDisplayName);
```

Comment:
Does this method only used in `doUpdateGroup `? If so, from where it says Group already exists error as if this method says group name is not changed, it does not perform a DB check, correct?

Reply:
Yes. This method is only used in `doUpdateGroup`. It doesn't give any group already exist error. But it will not update the DB with a log warn saying "There is no updated field in the group:"

Reply:
Got it thanks! Thought the issue mentioned in the PR description is directly related. Shall we update the description mentioning what would be the exact thing this PR is trying to fix? I think the mentioned issue is for a more general case and does not mention the part that we are fixing here